streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -16,224 +16,31 @@
-        this.logContext = new LogContext(logPrefix);
-        this.log = logContext.logger(getClass());
-    }
-
-    @Override
-    public TaskId id() {
-        return id;
-    }
-
-    @Override
-    public String applicationId() {
-        return applicationId;
-    }
+    private Task.State state = CREATED;
 
     @Override
-    public Set<TopicPartition> partitions() {
-        return partitions;
+    public final Task.State state() {
+        return state;
     }
 
-    @Override
-    public ProcessorTopology topology() {
-        return topology;
-    }
-
-    @Override
-    public ProcessorContext context() {
-        return processorContext;
-    }
-
-    @Override
-    public StateStore getStore(final String name) {
-        return stateMgr.getStore(name);
-    }
-
-    /**
-     * Produces a string representation containing useful information about a Task.
-     * This is useful in debugging scenarios.
-     *
-     * @return A string representation of the StreamTask instance.
-     */
-    @Override
-    public String toString() {
-        return toString("");
-    }
-
-    /**
-     * Produces a string representation containing useful information about a Task starting with the given indent.
-     * This is useful in debugging scenarios.
-     *
-     * @return A string representation of the Task instance.
-     */
-    public String toString(final String indent) {
-        final StringBuilder sb = new StringBuilder();
-        sb.append(indent);
-        sb.append("TaskId: ");
-        sb.append(id);
-        sb.append("\n");
-
-        // print topology
-        if (topology != null) {
-            sb.append(indent).append(topology.toString(indent + "\t"));
-        }
-
-        // print assigned partitions
-        if (partitions != null && !partitions.isEmpty()) {
-            sb.append(indent).append("Partitions [");
-            for (final TopicPartition topicPartition : partitions) {
-                sb.append(topicPartition.toString()).append(", ");
-            }
-            sb.setLength(sb.length() - 2);
-            sb.append("]\n");
-        }
-        return sb.toString();
-    }
-
-    /**
-     * Package-private for testing only
-     *
-     * @throws StreamsException If the store's change log does not contain the partition
-     */
-    void registerStateStores() {
-        if (topology.stateStores().isEmpty()) {
-            return;
+    final void transitionTo(final Task.State newState) {
+        final State oldState = state();
+        if (oldState == CREATED && (newState == RESTORING || newState == CLOSING)) {
+        } else if (oldState == RESTORING && (newState == RUNNING || newState == SUSPENDED || newState == CLOSING)) {
+        } else if (oldState == RUNNING && (newState == SUSPENDED || newState == CLOSING)) {
+        } else if (oldState == SUSPENDED && (newState == RUNNING || newState == CLOSING)) {
+        } else if (oldState == CLOSING && (newState == CLOSING || newState == CLOSED)) {
```

Comment:
This is a quite weird way to express a logical condition... Why not just use `||` ? Or even better, why not encode valid transitions in an `enum` type similar to existing code -> https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java#L137-L143

Reply:
I've moved it to `Task` following the similar pattern of KafkaStreams, StreamThread. LMK WDYT.

Reply:
Sounds good!